common/src/main/java/io/pravega/common/util/BlockingAsyncIterator.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.Synchronized;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * This class is a wrapper over async iterator that implements java.util.Iterator interface.
+ * This is blocking wrapper on async iterator's getNext method which implements blocking hasNext and next methods of 
+ * iterator interface. 
+ * This class is threadsafe but that has the drawback of blocking on asyncIterator's getNext within a lock. 
+ */
+@ThreadSafe
+@VisibleForTesting
+public class BlockingAsyncIterator<T> implements Iterator<T> {
```

Comment:
Just to be on the safe side, would we need unit tests for checking the functionality of `BlockingAsyncIterator` and `ContinuationTokenAsyncIterator`?